packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -150,6 +151,24 @@ String getGradleVersionForAndroidPlugin(Directory directory) {
   return getGradleVersionFor(androidPluginVersion);
 }
 
+const int _kExecPermissionMask = 0x8049; // u+x
+
+/// Returns [true] if [executable] has execute permission.
+bool _hasExecutePermission(File executable) {
+  final FileStat stat = executable.statSync();
+  assert(stat.type != FileSystemEntityType.notFound);
+  printTrace('${executable.path} mode: ${stat.mode} ${stat.modeString()}.');
+  return stat.mode & _kExecPermissionMask == _kExecPermissionMask;
+}
+
+/// Gives execute permission to [executable] if it doesn't have it already.
+void _giveExecutePermissionIfNeeded(File executable) {
+  if (!_hasExecutePermission(executable)) {
+    printTrace('Trying to give execute permission to ${executable.path}.');
+    os.makeExecutable(executable);
```

Comment:
This only does u+x, so this is going to run every time if it doesn't start out with a+x.

Reply:
`os.makeExecutable(..)` sets `a+x`

Reply:
Oh, yeah, you're right. Thanks.